thunder/core/prims.py

```diff
@@ -2375,6 +2376,29 @@ def real_meta(a: complex | TensorProxy) -> float | TensorProxy:
     meta=real_meta,
 )
 
+
+def imag_meta(a: complex | TensorProxy) -> float | TensorProxy:
+    utils.check_type(a, (TensorProxy, complex))
+    dtyp = dtypes.to_dtype(a, true_dtype=True)
+    utils.check(
+        dtyp, lambda: f"imag expected a complex tensor or number, but receive a tensor or number with dtype {dtyp}"
+    )
```

Comment:
I know this is based on the meta for `real`, but I don't understand this check either here or there.  It asserts that `dtyp` isn't falsey.  The output of `to_dtype` is either a dtype (never falsey, right?) or `None`.  But it can never be `None` because the the `check_type` in the first line of the function.  Based on the error message, should this instead be `utils.check(is_complex_dtype(dtyp), ...`?

Reply:
Yes! In general we shouldn't rely on implicit conversions to bool, anyway, because it can be confusing